langroid/agent/batch.py

```diff
@@ -19,66 +22,161 @@
 
 load_dotenv()
 
+T = TypeVar("T")
+U = TypeVar("U")
 
-def run_batch_tasks(
-    task: Task,
-    items: List[Any],
-    input_map: Callable[[Any], str | ChatDocument] = lambda x: str(x),
-    output_map: Callable[[ChatDocument | None], Any] = lambda x: x,
+
+def run_batch_task_gen(
```

Comment:
Nice, this is useful when we want to generate slightly different tasks in a batch

Reply:
Maybe worth adding a simple test in `test_batch.py`

Reply:
It's also nice in cases the cloning process wouldn't give a correct copy, say when the agent state is modified in some way